cmd/notation/cert_gen.go

```diff
@@ -109,39 +111,73 @@ func generateTestKey(bits int) (crypto.Signer, []byte, error) {
 	return key, keyPEM, nil
 }
 
-func generateTestSelfSignedCert(key crypto.Signer, hosts []string, expiry time.Duration) (*x509.Certificate, []byte, error) {
+func generateCert(template, parent *x509.Certificate, publicKey *rsa.PublicKey, privateKey *rsa.PrivateKey) (*x509.Certificate, []byte, error) {
```

Comment:
Just a question. The old code use `crypto.Signer` as a parameter, why do you change it to a concrete type `rsa.PublicKey`?  If we use an interface, then we can easily replace `rsa.GenerateKey(rand.Reader, bits)` with `ecdsa.GenerateKey(elliptic.P512(), rand.Reader)`.

Reply:
Good catch, I had the same consideration. But after a second thought, I think only using rsa is better here: according to cert generate-test's [Usage](https://github.com/notaryproject/notation/blob/bb5b49728798376a518b522f1693cf29b0376fa6/cmd/notation/cert.go#L61), it only generates RSA key as a test. So I think we should restrict the usage of this command and make it testing and demo purpose only?

Reply:
That makes sense :)